File: src/main/java/com/github/hdzitao/editstarters/EditStarters.java
package com.github.hdzitao.editstarters;

import com.github.hdzitao.editstarters.dependency.StarterInfo;

import java.util.Collection;

/**
 * 整个项目的顶级功能
 */
public interface EditStarters {
    /**
     * 删除starters
     */
    void removeStarters(Collection<StarterInfo> dependencies);

    /**
     * 添加starters
     */
    void addStarters(Collection<StarterInfo> dependencies);
}

File: src/main/kotlin/io/github/hdzitao/editstarters/EditStarters.kt
package io.github.hdzitao.editstarters

import io.github.hdzitao.editstarters.springboot.Starter


/**
 * 插件顶层接口
 *
 * @version 3.2.0
 */
interface EditStarters {
    /**
     * 添加starters
     */
    fun addStarters(dependencies: Collection<Starter>)

    /**
     * 删除starters
     */
    fun removeStarters(dependencies: Collection<Starter>)
}
